ocs_ci/deployment/ibm.py

```diff
@@ -0,0 +1,524 @@
+            # Create configmap cluster-monitoring-config to reconfigure
+            # telemeter server url when 'persistent-monitoring' is False
+            create_configmap_cluster_monitoring_pod(
+                telemeter_server_url=config.ENV_DATA["telemeter_server_url"])
+
+        # Change registry backend to OCS CEPHFS RWX PVC
+        registry.change_registry_backend_to_ocs()
+
+        # Verify health of ceph cluster
+        # TODO: move destroy cluster logic to new CLI usage pattern?
+        logger.info("Done creating rook resources, waiting for HEALTH_OK")
+        try:
+            ceph_health_check(namespace=self.namespace, tries=30, delay=10)
+        except CephHealthException as ex:
+            err = str(ex)
+            logger.warning(f"Ceph health check failed with {err}")
+            if "clock skew detected" in err:
+                logger.info(
+                    f"Changing NTP on compute nodes to"
+                    f" {constants.RH_NTP_CLOCK}"
+                )
+                update_ntp_compute_nodes()
+                assert ceph_health_check(
+                    namespace=self.namespace, tries=60, delay=10
+                )
+
+        # patch gp2/thin storage class as 'non-default'
+        self.patch_default_sc_to_non_default()
+
+        # Modify Noobaa endpoint auto scale values according to the cluster specs
+        # if check_nodes_specs(min_cpu=constants.MIN_NODE_CPU, min_memory=constants.MIN_NODE_MEMORY):
+        #     logger.info(
+        #         "The cluster specs meet the minimum requirements and "
+        #         "therefore, NooBaa auto scale will be enabled"
+        #     )
+        #     min_nb_eps = config.DEPLOYMENT.get('min_noobaa_endpoints')
+        #     max_nb_eps = config.DEPLOYMENT.get('max_noobaa_endpoints')
+        #     change_noobaa_endpoints_count(min_nb_eps=min_nb_eps, max_nb_eps=max_nb_eps)
+        # else:
+        #     logger.warning(
+        #         "The cluster specs do not meet the minimum requirements and "
+        #         "therefore, NooBaa auto scale will remain with its default values"
+        #     )
+        #     min_eps = 1
+        #     max_eps = 1 if float(config.ENV_DATA['ocs_version']) < 4.6 else 2
+        #     logger.info(
+        #         f"The Noobaa endpoint auto scale values: min: {min_eps}, max: {max_eps}"
+        #     )
+
+    def destroy_cluster(self, log_level="DEBUG"):
+        """
+        Base destroy cluster method, for more platform specific stuff please
+        overload this method in child class.
+
+        Args:
+            log_level (str): log level for installer (default: DEBUG)
+        """
+        if not config.ENV_DATA['skip_ocs_deployment']:
+            self.destroy_ocs()
+
+        if not config.ENV_DATA['skip_ocp_deployment']:
+            logger.info("Destroy of OCP not implemented yet.") 
+
+
+def get_device_paths(worker_names):
```

Comment:
get_device_paths needs to refer to config in the case of Power. It cannot go and query nodes and look for directories in /dev/disk/by-id etc..

Reply:
`get_device_paths` is currently only used from the `setup_local_storage` function in our deployment code. I think having a function here with the same name is just going to cause issues in troubleshooting later on. As the function is only two (pretty basic) lines of code, and doesn't actually use the `worker_names` argument anyways, I think we can safely remove the function and just perform those two lines where we are calling `get_device_paths` now. What do you think?

Reply:
I see now that the issue is you're also overwriting `get_local_storage`. This is another case where there is far too much duplication. This is also a standalone function that is defined in the base deployment module. If there is logic that needs to change based on the platform type, we need to change it there in the original definition of the function. If things are getting a bit too complicated, maybe we can consider breaking that function into a class that can handle the platform specific needs. For now, I think having the additional functionality that you need behind an `if` is sufficient. After looking at the diff between the original function and the one you have in `ibm.py` I don't think it will be too complicated to adjust the original `setup_local_storage` and `get_device_paths` functions to do what you need them to do if the platform type is IBM power.

TLDR: adjust the original functions in `deployment.py` to do what you need `if platform == "powervs"`.  This will reduce the amount of duplication between `ibm.py` and `deployment.py`.

Reply:
ok, I misunderstood earlier comments,  initially I modified deployment.py, and then reviewers (ebenahar) wanted me to create a subclass of Deployment into ibm.py. Now I understand that I should do both with minimal duplication and it is ok to change deployment.py with conditional coding. I will do and raise another version of this pull request.

Reply:
You can absolutely (and should) create a subclass and override/add any methods you need to regarding the OCP deployment. Please check out the [azure](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/azure.py) class for a good example of this.

Since the OCS deployment is (mostly) the same across platforms, this is handled in the base class. If there is a slight change in logic for a particular platform we can change it there and not have to copy the entire ocs deployment method. Same with the `setup_local_storage` and `get_device_paths` functions. Since these are standalone functions in that module and not a part of the class itself it's easier to just have them handle the logic change based on platform.

Feel free to ping me if you have any questions.

Reply:
This is resolved now, removed every common method from ibm.py.